### PR TITLE
cmd/{k8s-operator, tailscaled},envknob,kube:set app name for Kubernetes Operator proxies

### PR DESCRIPTION
**Pull request by irbekrm**
Updates tailscale/corp#22920

Sets app type for Kubernetes Operator proxies:

L3 ingress proxy (including ExternalName svc) - "k8s-operator-ingress-proxy"
L7 ingress proxy  -  "k8s-operator-ingress-resource"
egress proxy   -  "k8s-operator-egress-proxy"
Connector resource     - "k8s-operator-connector-resource"

### CODE REVIEW

File: kube/metrics.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+	// Clientmetrics for Tailscale Kubernetes Operator components
+	IngressProxyCount              = "k8s_ingress_proxies"   // L3
+	IngressResourceCount           = "k8s_ingress_resources" // L7
+	EgressProxyCount               = "k8s_egress_proxies"
+	ConnectorResourceCount         = "k8s_connector_resources"
+	ConnectorWithSubnetRouterCount = "k8s_connector_subnetrouter_resources"
+	ConnectorWithExitNodeCount     = "k8s_connector_exitnode_resources"
+	NameserverCount                = "k8s_nameserver_resources"
+)
```
Comment (irbekrm) [inline kube/metrics.go:23]: The client metrics are not new in this PR, just moved them around to make it easier to see all metric types for operator components

File: kube/metrics.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package kube
+
+const (
+	// Hostinfo App values for the Tailscale Kubernetes Operator components.
+	AppOperator        = "k8s-operator"
+	AppAPIServerProxy  = "k8s-operator-proxy"
```
Comment (irbekrm) [inline kube/metrics.go:9]: Pre-existing app types, just moved around

File: kube/metrics.go (hunk 1 above)
Comment (maisem) [reply kube/metrics.go:23]: Maybe prefix them with Metric like you prefix App on the other ones

File: cmd/containerboot/main.go [hunk 3]
```diff
@@ -201,7 +204,9 @@ func main() {
 			log.Fatalf("error setting up for running on Kubernetes: %v", err)
 		}
 	}
-
+	if cfg.App != "" {
+		hostinfo.SetApp(cfg.App)
```
Comment (maisem) [inline cmd/containerboot/main.go:208]: This won't work here as this is in containerboot and you need to set this in tailscaled.

File: cmd/tailscale/depaware.txt [hunk 4]
```diff
@@ -98,6 +98,7 @@ tailscale.com/cmd/tailscale dependencies: (generated by github.com/tailscale/dep
         tailscale.com/internal/noiseconn                             from tailscale.com/cmd/tailscale/cli
         tailscale.com/ipn                                            from tailscale.com/client/tailscale+
         tailscale.com/ipn/ipnstate                                   from tailscale.com/client/tailscale+
+        tailscale.com/kube                                           from tailscale.com/envknob
```
Comment (irbekrm) [inline cmd/tailscale/depaware.txt:101]: Hopefully this is OK, that package contains only some types and is intended for sharing with things that don't want to pull in upstream kube libraries

File: cmd/containerboot/main.go (hunk 3 above)
Comment (irbekrm) [reply cmd/containerboot/main.go]: I've fixed this and tested that it now is actually set as expected

File: cmd/containerboot/main.go [hunk 5]
```diff
@@ -201,7 +201,6 @@ func main() {
 			log.Fatalf("error setting up for running on Kubernetes: %v", err)
 		}
 	}
```
Comment (maisem) [inline cmd/containerboot/main.go:203]: empty change? remove?

Comment (irbekrm) [reply cmd/containerboot/main.go]: done

**Review by maisem — APPROVED** [review]

**Pull request merged**